interface/modules/custom_modules/oe-module-dorn/public/route_edit.php

```diff
@@ -0,0 +1,132 @@
+ * @copyright Copyright (c) 2022 Brad Sharp <brad.sharp@claimrev.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+ require_once "../../../../globals.php";
+
+ use OpenEMR\Common\Acl\AclMain;
+ use OpenEMR\Common\Csrf\CsrfUtils;
+ use OpenEMR\Common\Twig\TwigContainer;
+ use OpenEMR\Core\Header;
+ use OpenEMR\Modules\Dorn\ConnectorApi;
+ use OpenEMR\Modules\Dorn\models\CreateRouteFromPrimaryViewModel;
+ use OpenEMR\Modules\Dorn\DisplayHelper;
+ use OpenEMR\Modules\Dorn\LabRouteSetup;
+ use OpenEMR\Modules\Dorn\AddressBookAddEdit;
+
+$labGuid = "";
+$message = "";
+
+if (!empty($_GET)) {
+    if (!CsrfUtils::verifyCsrfToken($_GET["csrf_token_form"])) {
+        CsrfUtils::csrfNotVerified();
+    }
+}
+
+if (!empty($_POST)) {
+    if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"])) {
+        CsrfUtils::csrfNotVerified();
+    }
+    //lets lookup the lab information we want to add a route for!
+    $routeData = CreateRouteFromPrimaryViewModel::loadByPost($_POST);
+    $apiResponse =  ConnectorApi::createRoute($routeData);//create the route on dorn, we have all we need to do so
+    if ($apiResponse->isSuccess) {
+        $ppid = 0;
+        $uid = 0;
+        $labData = ConnectorApi::getLab($routeData->labGuid);
+        $note = "labGuid:" . $labData->labGuid;
+
+        $setupRouteInfo = LabRouteSetup::getRouteSetup($apiResponse->labGuid, $apiResponse->routeGuid);
+        if ($setupRouteInfo != null) {
+            $ppid = $setupRouteInfo["ppid"];
+            $uid = $setupRouteInfo["uid"];
+        }
+
+        //we've added this lab to the address book here.
+        $uid = AddressBookAddEdit::createOrUpdateRecordInAddressBook($uid, $labData->name, $labData->address1, $labData->address2, $labData->city, $labData->state, $labData->zipCode, $labData->Website, $labData->phoneNumber, $labData->faxNumber, $note);
+        $ppid = LabRouteSetup::createUpdateProcedureProviders($ppid, $labData->name, $routeData->npi, $labData->labGuid, $uid);
+
+        //lets add/update to the new dorn route table
+        $isLabSetup = LabRouteSetup::createDornRoute($apiResponse->labName, $apiResponse->routeGuid, $apiResponse->labGuid, $ppid, $uid, $labData->textLineBreakCharacter, $routeData->labAccountNumber);
+        if ($isLabSetup) {
+            $message = "Lab has been setup";
+        } else {
+            $message = "Failure creating route!";
+        }
+    } else {
+        if ($apiResponse->responseMessage) {
+            $message = $apiResponse->responseMessage;
+        } else {
+            $message = "Error creating route, no information came back though";
+        }
+    }
+} else {
+    if (!empty($_GET)) {
+        $labGuid = $_REQUEST['labGuid'];
+    }
+}
+
+if (!AclMain::aclCheckCore('admin', 'users')) {
+    echo (new TwigContainer(null, $GLOBALS['kernel']))->getTwig()->render('core/unauthorized.html.twig', ['pageTitle' => xl("Edit/Add Procedure Provider")]);
+    exit;
+}
```

Comment:
- seeing same block of code for acl for all scripts with incorrect pageTitle
- also should have access controls protect entire script (ie. can still do stuff above is POST is populated)